theano/gpuarray/tests/test_opt.py

```diff
@@ -359,24 +359,19 @@ def test_pdbbreakpoint_op():
     assert _check_stack_trace(f)
 
 
-def test_local_gpu_elemwise_careduce():
-    x = theano.tensor.matrix()
-    o = (x * x).sum()
-    f = theano.function([x], o, mode=mode_with_gpu)
-    topo = f.maker.fgraph.toposort()
-    assert len(topo) == 3
-    assert topo[1].op.pre_scalar_op == theano.scalar.sqr
-    assert _check_stack_trace(f)
-    data = np.random.rand(3, 4).astype(theano.config.floatX)
-    utt.assert_allclose(f(data), (data * data).sum())
-
-    o = (x * x).sum(axis=1)
-    f = theano.function([x], o, mode=mode_with_gpu)
-    topo = f.maker.fgraph.toposort()
-    assert len(topo) == 3
-    assert topo[1].op.pre_scalar_op == theano.scalar.sqr
-    assert _check_stack_trace(f)
-    utt.assert_allclose(f(data), (data * data).sum(axis=1))
```

Comment:
Maybe we should keep that test in addition to the new one, but disable dnn optimizations in the mode.
That way we make sure the old optimization still works when there is no cudnn.